gqcp/tests/QCMethod/CC/QCMethod_CCD_test.cpp

```diff
@@ -0,0 +1,91 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-GQCP.  If not, see <http://www.gnu.org/licenses/>.
+
+#define BOOST_TEST_MODULE "CCD"
+
+#include <boost/test/unit_test.hpp>
+
+#include "Basis/transform.hpp"
+#include "Operator/SecondQuantized/SQHamiltonian.hpp"
+#include "QCMethod/CC/CCD.hpp"
+#include "QCMethod/CC/CCSDEnvironment.hpp"
+#include "QCMethod/CC/CCDSolver.hpp"
+#include "QCMethod/HF/RHF/DiagonalRHFFockMatrixObjective.hpp"
+#include "QCMethod/HF/RHF/RHF.hpp"
+#include "QCMethod/HF/RHF/RHFSCFSolver.hpp"
+
+
+/**
+ *  Check if the implementation of spinor-CCSD is correct, by comparing with a reference by crawdad (https://github.com/CrawfordGroup/ProgrammingProjects/tree/master/Project%2305).
+ *
+ *  The system under consideration is H2O in an STO-3G basisset.
+ */
+BOOST_AUTO_TEST_CASE(h2o_crawdad) {
+
+    // Prepare the canonical RHF spin-orbital basis.
+    const auto molecule = GQCP::Molecule::ReadXYZ("data/h2o_crawdad.xyz");
+    const auto N = molecule.numberOfElectrons();
+
+    GQCP::RSpinorBasis<double, GQCP::GTOShell> r_spinor_basis {molecule, "STO-3G"};
+    const auto r_sq_hamiltonian = GQCP::SQHamiltonian<double>::Molecular(r_spinor_basis, molecule);  // in an AO basis
+
+    auto rhf_environment = GQCP::RHFSCFEnvironment<double>::WithCoreGuess(N, r_sq_hamiltonian, r_spinor_basis.overlap().parameters());
+    auto plain_rhf_scf_solver = GQCP::RHFSCFSolver<double>::Plain();
+    const GQCP::DiagonalRHFFockMatrixObjective<double> objective {r_sq_hamiltonian};
+    const auto rhf_qc_structure = GQCP::QCMethod::RHF<double>().optimize(objective, plain_rhf_scf_solver, rhf_environment);
+    const auto rhf_parameters = rhf_qc_structure.groundStateParameters();
+
+    r_spinor_basis.transform(rhf_parameters.coefficientMatrix());
+
+
+    // Check if the intermediate RHF results are correct. We can't continue if this isn't the case.
+    const auto rhf_energy = rhf_qc_structure.groundStateEnergy() + GQCP::Operator::NuclearRepulsion(molecule).value();
+    const double ref_rhf_energy = -74.942079928192;
+    BOOST_REQUIRE(std::abs(rhf_energy - ref_rhf_energy) < 1.0e-09);
+
+
+    // Create a GSpinorBasis since we have implement spinor-CCD, and quantize the molecular Hamiltonian in it.
+    const auto g_spinor_basis = GQCP::GSpinorBasis<double, GQCP::GTOShell>::FromRestricted(r_spinor_basis);
+    const auto M = g_spinor_basis.numberOfSpinors();
+    const auto g_sq_hamiltonian = GQCP::SQHamiltonian<double>::Molecular(g_spinor_basis, molecule);  // in the canonical restricted spin-orbitals
+
+    // Use a manually-made orbital space (#FIXME in develop).
+    const auto orbital_space = GQCP::OrbitalSpace({0, 1, 2, 3, 4, 7, 8, 9, 10, 11}, {5, 6, 12, 13});  // occupied and virtual indices
```

Comment:
Just an FYI: check `QCMethod_CCSD_test.cpp` for a better, programmatic, way to create the required `OrbitalSpace`.

Reply:
@lelemmen https://github.com/GQCG/GQCP/blob/develop/gqcp/tests/QCMethod/CC/QCMethod_CCSD_test.cpp uses this way if I'm not mistaken?

Reply:
I didn't realize I hadn't pushed a branch I am working on. This 'new way' can be found in #583's `QCMethod_CCSD_test.cpp`.

Reply:
I've put the new code in the comments. If #583 is merged, I'll change my code. Should I review this new PR?

Reply:
My apologies, I was under the assumption that `develop` also contained the required API changes to incorporate this behavior. Indeed, if #583 is merged, you can update your test case to accommodate these changes.